specs/RasterizationScale.md

```diff
@@ -0,0 +1,164 @@
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+
+# Description
+To allow the app to control the DPI scale of WebView, we are adding a `RasterizationScale` property to the CoreWebView2Controller interface.
+The RasterizationScale property controls the DPI scaling for all UI in WebView.
+
+In order to maintain compatibility with apps developed before this API existed,
+WebView2 continues to detect monitor scale changes by default and will update the RasterizationScale property.
+When the RasterizationScale property is updated, the `RasterizationScaleChanged` event is raised.
+The app can tell WebView2 to stop updating the RasterizationScale property by changing
+`ShouldDetectMonitorDpiScaleChanges` from the default value of true, to false.
+
+# Examples
+The following code snippets show how to use the RasterizationScale and ShouldDetectMonitorScaleChanges properties,
+and now to listen to the RasterizationScaleChanged event.
+## Win32 C++
+```cpp
+void ViewComponent::SetRasterizationScale(float additionalScale)
+{
+    CHECK_FAILURE(m_controller->put_ShouldDetectMonitorScaleChanges(FALSE));
+    m_webviewAdditionalRasterizationScale = additionalScale;
+    // RasterizationScale is typically the monitor DPI scale and text scaling, but
+    // the app could add an additional scale for its own scenarios.
+    double rasterizationScale =
+        additionalScale * m_appWindow->GetDpiScale() * m_appWindow->GetTextScale();
+    CHECK_FAILURE(m_controller->put_RasterizationScale(rasterizationScale));
+}
+
+Callback<ICoreWebView2RasterizationScaleChangedEventHandler>(
+    [this](ICoreWebView2Controller* sender, IUnknown* args) -> HRESULT {
+        double rasterizationScale;
+        CHECK_FAILURE(m_controller->get_RasterizationScale(&rasterizationScale));
+
+        std::wstring message = L"WebView2APISample (RasterizationScale: " +
+            std::to_wstring(int(rasterizationScale * 100)) + L"%)";
+        SetWindowText(m_appWindow->GetMainWindow(), message.c_str());
+        return S_OK;
+    })
+.Get(), &m_rasterizationScaleChangedToken));
+```
+## .Net
+```c#
+public void SetRasterizationScale(double additionalScale)
+{
+    CoreWebView2Controller.ShouldDetectMonitorScaleChanges = false;
+    m_webviewAdditionalRasterizationScale = additionalScale;
+    
+    // RasterizationScale is typically the monitor DPI scale and text scaling, but
+    // the app could add an additional scale for its own scenarios.
+    double rasterizationScale = additionalScale * GetDpiScale() * GetTextScale();
+    CoreWebView2Controller.RasterizationScale = rasterizationScale;
+}
+```
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## Win32 C++
+```c#
+interface ICoreWebView2Controller2 : ICoreWebView2Controller {
+  /// The rasterization scale for the WebView. The rasterization scale is the
+  /// combination of the monitor DPI scale and text scaling set by the user.
```

Comment:
Is this the standard definition of rasterization scale? Does this match [XamlRoot.RasterizationScale](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.XamlRoot.RasterizationScale)?

Reply:
I'm not sure if XamlRoot.RasterizationScale includes text scaling. It may also include additional Shell scaling (on some skus). When we discussed this with the composition team, they suggested calling this property rasterization scale. The expectation is the XAML framework will take the XamlRoot.RasterizationScale times any additional scale in the XAML tree, and set that on CoreWebView2 as the RasterizationScale for WebView2.